### ISSUE TITLE
Thread in encalve

### ISSUE THREAD

**Issue by hyunsooda** [openenclave/openenclave#3859]
In order to achieve better performance, we can trigger many ecalls simultaneously in host-side and host threads should be created before. I'm currently using this technic but, it is not uncomfortable from point of a developer using OE since it makes programmers write additional codes like below.

1. Generate threads in host-side (OCall definition is required)
2. Corresponding ecalls are required so that host threads created can do ecall.
3. The worst thing, when host thread does ecall, it should have some data to be used for task to invoking ecall but, people may do not want to pass data from enclave into host-side as it's almost sensitive, so it seems that programmers should make global variables in enclave. But, there is an alternative method like encryption and decryption but it seems also costly or overengineering.

I don't know why OE [removed function creating thread](https://github.com/openenclave/openenclave/blob/09aa94bf748fca1a9a2581abadf87e52db307c0e/enclave/core/pthread.c#L58) like ```pthread_create()``` or ```std::thread()```. Does making a thread in enclave require additional something technically and internally?

Thank you

**Comment by anakrish** [comment]
Related to #3697

**Comment by hyunsooda** [comment]
Hi @anakrish,
Thanks for the useful link. The proposal is not implemented in the latest version and seems to need more review. Is there a plan to complete that proposal in the near future?

**Comment by radhikaj** [comment]
Yes, we agree that this would be useful. No one is working on this right now. Would you be able to work with us to contribute this?
@anakrish believes that adding pthread_create and pthread_join would enable us to port more applications.

**Comment by radhikaj** [comment]
Duplicate of https://github.com/openenclave/openenclave/issues/4253. See for more details.

**Comment by xanderdunn** [comment]
This functionality would be hugely beneficial. Currently only the less efficient multi-enclave approach described [here](http://madsys.cs.tsinghua.edu.cn/publications/ApSys18-shen.pdf) is possible with OpenEnclave. The more performant single-enclave multi-process approach used by [Oculum](https://github.com/occlum/occlum) is not possible here.
